python/pyarrow/tests/test_dataset.py

```diff
@@ -373,141 +357,70 @@ def test_partitioning():
+                 ds.field('i64') > 5, ds.field('i64') == 5,
+                 ds.field('i64') == 7]
     for expr in all_exprs:
+        print(str(expr))
+        assert isinstance(expr, ds.Expression)
         restored = pickle.loads(pickle.dumps(expr))
         assert expr.equals(restored)
 
 
-def test_expression_ergonomics():
+def test_expression_construction():
     zero = ds.scalar(0)
     one = ds.scalar(1)
     true = ds.scalar(True)
     false = ds.scalar(False)
     string = ds.scalar("string")
     field = ds.field("field")
 
-    assert one.equals(ds.ScalarExpression(1))
-    assert zero.equals(ds.ScalarExpression(0))
-    assert true.equals(ds.ScalarExpression(True))
-    assert false.equals(ds.ScalarExpression(False))
-    assert string.equals(ds.ScalarExpression("string"))
-    assert field.equals(ds.FieldExpression("field"))
-
-    expected = ds.AndExpression(ds.ScalarExpression(1), ds.ScalarExpression(0))
-    for expr in [one & zero, 1 & zero, one & 0]:
-        assert expr.equals(expected)
-
-    expected = ds.OrExpression(ds.ScalarExpression(1), ds.ScalarExpression(0))
-    for expr in [one | zero, 1 | zero, one | 0]:
-        assert expr.equals(expected)
-
-    comparison_ops = [
-        (operator.eq, ds.CompareOperator.Equal),
-        (operator.ne, ds.CompareOperator.NotEqual),
-        (operator.ge, ds.CompareOperator.GreaterEqual),
-        (operator.le, ds.CompareOperator.LessEqual),
-        (operator.lt, ds.CompareOperator.Less),
-        (operator.gt, ds.CompareOperator.Greater),
-    ]
-    for op, compare_op in comparison_ops:
-        expr = op(zero, one)
-        expected = ds.ComparisonExpression(compare_op, zero, one)
-        assert expr.equals(expected)
-
+    expr = zero | one == string
     expr = ~true == false
-    expected = ds.ComparisonExpression(
-        ds.CompareOperator.Equal,
-        ds.NotExpression(ds.ScalarExpression(True)),
-        ds.ScalarExpression(False)
-    )
-    assert expr.equals(expected)
-
     for typ in ("bool", pa.bool_()):
         expr = field.cast(typ) == true
-        expected = ds.ComparisonExpression(
-            ds.CompareOperator.Equal,
-            ds.CastExpression(ds.FieldExpression("field"), pa.bool_()),
-            ds.ScalarExpression(True)
-        )
-        assert expr.equals(expected)
 
     expr = field.isin([1, 2])
-    expected = ds.InExpression(ds.FieldExpression("field"), pa.array([1, 2]))
-    assert expr.equals(expected)
 
     with pytest.raises(TypeError):
-        field.isin(1)
+        expr = field.isin(1)
 
     # operations with non-scalar values
     with pytest.raises(TypeError):
-        field == [1]
+        expr = field == [1]
 
     with pytest.raises(TypeError):
-        field != {1}
+        expr = field != {1}
 
     with pytest.raises(TypeError):
-        field & [1]
+        expr = field & [1]
 
     with pytest.raises(TypeError):
-        field | [1]
+        expr = field | [1]
+
+    assert expr is not None  # silence flake8
```

Comment:
if you don't assign them to `expr` in the lines above (how it was before), you shouldn't need this to silence flake8. 
Or did you get another linter error when not assigning them to a variable?

Reply:
I didn't try that